src/main/java/seedu/jarvis/model/masterycheck/MasteryCheckList.java

```diff
@@ -0,0 +1,80 @@
+package seedu.jarvis.model.masterycheck;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.jarvis.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.jarvis.model.masterycheck.exceptions.DuplicateMasteryCheckException;
+import seedu.jarvis.model.masterycheck.exceptions.MasteryCheckNotFoundException;
+
+
+public class MasteryCheckList {
+    private final ObservableList<MasteryCheck> internalList = FXCollections.observableArrayList();
+
+    /**
+     * Adds a mastery check to the list.
+     */
+    public void add(MasteryCheck toAdd) {
+        requireNonNull(toAdd);
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Returns true if the list contains an equivalent mastery check as the given argument.
+     */
+    public boolean contains(MasteryCheck toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameConsultation);
+    }
+
+    public void setMasteryChecks(List<MasteryCheck> masteryChecks) {
+        requireNonNull(masteryChecks);
+        this.internalList.setAll(masteryChecks);
+    }
+    /*
+    public void setMasteryCheck(List<MasteryCheck> masteryChecks) {
+        requireNonNull(masteryChecks);
+        this.internalList.setAll(masteryChecks);
+    }*/
+
+    public ObservableList<MasteryCheck> asObservableList() {
+        return internalList;
+    }
+
+    public boolean isMasteryCheckInList(String name) {
+        return internalList.stream().anyMatch(masteryCheck -> masteryCheck.getIdentifier().equals(name));
+    }
+
+    /**
```

Comment:
is the type suppose to be Mastery Check and params suppose to be editedMasteryCheck?

Reply:
Hmm, I already have another method for that. Seems like I actually don't need this method at all! Took a bit of time but I deleted all of its uses. Thanks for reminding me!